runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/graph/CreateExecutableStageNodeFunction.java

```diff
@@ -195,38 +200,8 @@ public Node apply(MutableNetwork<Node, Edge> input) {
         Iterables.filter(input.nodes(), InstructionOutputNode.class)) {
       InstructionOutput instructionOutput = node.getInstructionOutput();
 
-      // If this is the input PCollection or the output PCollection for an ExecutableStage, it's
-      // necessary to check whether the window coder is not a GlobalWindow coder.
-      if (isExecutableStageInputPCollection(input, node)
-          || isExecutableStageOutputPCollection(input, node)) {
-        Coder<?> javaCoder =
-            CloudObjects.coderFromCloudObject(CloudObject.fromSpec(instructionOutput.getCodec()));
-        // For now, Dataflow runner harness only deal with FixedWindow.
-        if (FullWindowedValueCoder.class.isInstance(javaCoder)) {
-          FullWindowedValueCoder<?> windowedValueCoder = (FullWindowedValueCoder<?>) javaCoder;
-          Coder<?> windowCoder = windowedValueCoder.getWindowCoder();
-          if (IntervalWindowCoder.class.isInstance(windowCoder)) {
-            fakeWindowingStrategyId = "generatedFixedWindowingStrategy" + idGenerator.getId();
-            try {
-              // Since the coder is the only needed from a WindowingStrategy, the size field of one
-              // FixedWindows is meaningless here.
-              RunnerApi.MessageWithComponents windowingStrategyProto =
-                  WindowingStrategyTranslation.toMessageProto(
-                      WindowingStrategy.of(FixedWindows.of(Duration.standardSeconds(1))),
-                      sdkComponents);
-              componentsBuilder.putWindowingStrategies(
-                  fakeWindowingStrategyId, windowingStrategyProto.getWindowingStrategy());
-              componentsBuilder.putAllCoders(windowingStrategyProto.getComponents().getCodersMap());
-              componentsBuilder.putAllEnvironments(
-                  windowingStrategyProto.getComponents().getEnvironmentsMap());
-            } catch (IOException exc) {
-              throw new RuntimeException("Could not convert FixedWindow stratey to proto", exc);
-            }
-          }
-        }
-      }
-
       String coderId = "generatedCoder" + idGenerator.getId();
+      String windowingStrategyId = fakeGlobalWindowingStrategyId;
```

Comment:
I would prefer being defensive and leave this unset, then below setting it in each of the conditions (including adding a check for global windows and the "Coder probably wasn't a java coder" exception clause. This will make it clearer when we're right and when we're just sticking something in.

Reply:
I don't think we should have an exception clause if this is not a java coder. In my latest changes, I set it to use GlobalWindow if it's not a java coder.